### PR TITLE
[JENKINS-43669] - Fix incorrect serialization of private keys (regression in 1.6)

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-43669 fix in #10 caused regressions in the module. It looks like the serialized objects are different under the change, Jenkins master does not correctly resolve the private certificate after the change.

- [x] - Reproduce the issue in test
- [x] - Fix the issue

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Deployed the patch as 1.7-20190715.235542-1 to get the CI running for https://github.com/jenkinsci/jenkins/pull/4111

File: src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/PublicKeySignatureWriter.java [hunk 1]
```diff
@@ -45,12 +47,12 @@ public String asString(DSAPublicKey key) {
         try {
             ByteArrayOutputStream output = new ByteArrayOutputStream();
             DSAParams keyParams = Objects.requireNonNull(key.getParams(), "No DSA params available");
-            KeyEncodeHelper.encodeString(output, "ssh-dss", StandardCharsets.UTF_8);
+            KeyEncodeHelper.encodeString(output, "ssh-dss", StandardCharsets.ISO_8859_1);
```
Comment (fcojfernandez) [inline src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/PublicKeySignatureWriter.java]: Why did you have to change the charset?

Comment (varyvol) [inline src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/PublicKeySignatureWriter.java]: Is this valid for the future too?

Comment (oleg-nenashev) [reply src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/PublicKeySignatureWriter.java]: This is how the original code behaved before #10 . I do not have opinion about future, just fixing the known regression

Comment (oleg-nenashev) [reply src/main/java/org/jenkinsci/main/modules/cli/auth/ssh/PublicKeySignatureWriter.java]: I reverted it to the one used in TrileadSSH. Just to keep the implementations as close as possible

**Comment by oleg-nenashev** [comment]
Will merge it to cut a new release for the downstream PR.
The new release will not be worse than the previous one so 🤷‍♂ . Will address any comments in the follow-up

**Pull request merged**